internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -895,6 +897,41 @@ func updateRequest(ctx context.Context, d *schema.ResourceData, projectID, clust
 	return cluster, nil
 }
 
+func syncAutoScalingConfigs(replicationSpecs *[]admin.ReplicationSpec20250101) {
```

Comment:
Would capture in a comment why we need to do this sync on our side. From my understanding this is due to:
- Existing replication specs can have the autoscaling values present in the state with default values even if not defined in the config, while a new replication spec would not have this value present. 
- API expects autoScaling and analyticsAutoScaling aligned cross all region configs.

Correct me if the above statements are not correct.

Reply:
Correct.